### PR TITLE
Feature: Typescript definition files

### PR DESCRIPTION
**Pull request by JuliaABurch**
## Description
This commit fixes up the JSDocs for the `@amazon-sumerian-hosts/babylon` package, primarily by `typedef`ing several object structures used throughout `HostObject`, and standardizing the way we refer to types from external modules (AWS SDK, BabylonJS, etc). Typescript unfortunately doesn't support the `external` tag, and will generate nonsense from it.

Typescript is added as a dev-dependency, along with a build target that will compile definition files from the JSDoc documentation. 

To add type definitions for the other submodules, a `tsconfig.json` file needs to be added at the root of that submodule (it can largely be copied from the `tsconfig.json` file defined for the `@amazon-sumerian-hosts/babylon` package.) Then a project reference needs to be added to the monorepo's `tsconfig.json`, so that the compiler knows where to find those nested `tsconfig.json` files.

## Reviewer Testing Instructions
- Run `npm run compile-ts` to generate the `*.d.ts` definition files
- When the `@amazon-sumerian-hosts/babylon` package is then installed and imported into a Typescript project (such as one generated by the BabylonJS editor), there should be typechecking and intellisense without any further configuration needed by that project

## Submission Checklist
I confirm that I have...
- [x] removed hard-coded Cognito IDs
- [x] manually smoke-tested the BabylonJS integration tests
- [x] manually smoke-tested the BabylonJS demos
- [x] manually smoke-tested the Three.js integration tests
- [x] manually smoke-tested the Three.js demo

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amazon-sumerian-hosts-babylon/src/Babylon.js/HostObject.js [hunk 1]
```diff
@@ -613,7 +646,9 @@ class HostObject extends CoreHostObject {
   }
 
   /**
-   * @private
+   * Reads JSON file
+   * @param {string} url
+   * @returns
```
Comment (eherozhao) [inline packages/amazon-sumerian-hosts-babylon/src/Babylon.js/HostObject.js:651]: nit: might miss the description of returns here.

File: packages/amazon-sumerian-hosts-babylon/src/Babylon.js/HostObject.js [hunk 2]
```diff
@@ -693,7 +711,7 @@ class HostObject extends CoreHostObject {
    * parameter. This can be used to create one of the built-in hosts or your own
    * custom host.
    *
-   * When creating a custom host, use H`OST.HostObject.getCharacterConfig()` to
```
Comment (eherozhao) [inline packages/amazon-sumerian-hosts-babylon/src/Babylon.js/HostObject.js:696]: ah what a typo. We didn't even notice it.

File: packages/amazon-sumerian-hosts-babylon/src/Babylon.js/PointOfInterestFeature.js [hunk 3]
```diff
@@ -23,6 +23,10 @@ class PointOfInterestFeature extends CorePointOfInterestFeature {
     return obj instanceof TransformNode;
   }
 
+  /**
+   *
```
Comment (eherozhao) [inline packages/amazon-sumerian-hosts-babylon/src/Babylon.js/PointOfInterestFeature.js:27]: nit: Do we need a description for the function here? If not, I think we can remove line 27.

File: packages/amazon-sumerian-hosts-babylon/src/Babylon.js/PointOfInterestFeature.js [hunk 4]
```diff
@@ -32,6 +36,10 @@ class PointOfInterestFeature extends CorePointOfInterestFeature {
     this.target = this._scene.getTransformNodeByName(name);
   }
 
+  /**
+   *
```
Comment (eherozhao) [inline packages/amazon-sumerian-hosts-babylon/src/Babylon.js/PointOfInterestFeature.js:40]: The same as above.

**Review by eherozhao — APPROVED** [review]

**Review by runpiw — APPROVED** [review]

**Pull request merged**